### PR TITLE
update README for non-multi Tenancy

### PR DESCRIPTION
**Pull request by harrisonchen201703**
- Add usage for non-multi-Tenancy users
- Remove "task" label from Readme since task may not be executed independently
- Add description for TenantID/Token variable

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +26,22 @@ There are no dependencies.
 
 Below is a sample playbook:
 
+##### For deploying agents to Non-Multi-Tenancy Deep Security Manager
+```yaml
+- hosts: all
+  roles:
+    - role: deep-security.deep-security-agent
+      action: deploy
+      dsm_agent_download_hostname: app.deepsecurity.trendmicro.com
```
Comment (chchhsiao) [inline README.md]: I think we should not use DSaaS url as example, I think use the default OP setting, ex. 4119 and 4120 port will be better to fit users environment.

File: README.md [hunk 2]
```diff
@@ -26,6 +26,22 @@ There are no dependencies.
...
+      dsm_agent_download_port: 443
+      dsm_agent_activation_hostname: agents.deepsecurity.trendmicro.com
+      dsm_agent_activation_port: 443
+      policy_id: 1
+      group_id: 1
+      force_reactivation: false
+```
+
+##### For deploying agents to Multi-Tenancy Deep Security Manager
```
Comment (chchhsiao) [inline README.md]: Maybe we can add ", ex. Deep Security As A Service" at the end.

File: README.md [hunk 3]
```diff
@@ -26,6 +26,22 @@ There are no dependencies.
 
 Below is a sample playbook:
```
Comment (chchhsiao) [inline README.md]: As we added a new sample, we might need to alter this like "Below are sample usages for playbook:"

File: README.md [hunk 4]
```diff
@@ -59,14 +73,13 @@ dsm_agent_download_hostname | String | The hostname of the Deep Security Manager
...
 dsm_agent_activation_hostname | String | The hostname for the agents to communicate with once deployed. For Marketplace and software deployments, this is typically the same hostname as 'dsm_agent_download_hostname'. | agents.deepsecurity.trendmicro.com
 dsm_agent_activation_port | Int | The port used for the agent heartbeat (the regular communication). For Marketplace and software deployments, the default is 4120. | 443
-tenant_id | String | In a multi-tenant installation (like Deep Security as a Service), this identifies the tenant account to register the agent with. |
-token/tenant_password | String | In a multi-tenant installation (like Deep Security as a Service), this identifies the tenant account to register the agent with. In latest Deep Security Manager, "tenant_password" has been replaced with "token". "tenant_password" has been kept for backwards compatibility |
-policy_id | String | The Deep Security ID assigned to the policy and applied to the agents on activation. |
+tenant_id | String | In a multi-tenant installation (like Deep Security as a Service), this identifies the tenant account to register the agent with. | Only required in multi-tenancy environment
```
Comment (chchhsiao) [inline README.md]: The requirement is not default value, should be merged as part of description.
If you'd like to high-light this, maybe we can use BOLD style.

File: README.md [hunk 5]
```diff
@@ -47,8 +63,6 @@ Please refer to the [Variables](#variables) section for required variables.
 
 ## Variables<a id="variables"></a>
 
-
-#### Task : deploy.yml
 ###### action: deploy
```
Comment (chchhsiao) [inline README.md]: As tasks were removed, action should be #### but not ######

File: README.md [hunk 6]
```diff
@@ -47,8 +63,6 @@ Please refer to the [Variables](#variables) section for required variables.
 
 ## Variables<a id="variables"></a>
```
Comment (chchhsiao) [inline README.md]: Maybe we should consider to put a action list here first as all the variable will be binding with action to be used.
And we might remove the duplicated definition for the variable items.

File: README.md [hunk 7]
```diff
@@ -93,43 +105,28 @@ group_id | String | The Deep Security ID assigned to the computer group and appl
...
-      
-- name: "Set policy for webservers"
-  hosts: web
-  roles:
-    - role: deep-security.deep-security-agent
-      action: set-policy-by-name
-      policy_name: MyCompany - Webserver            
-```
+
+##### Action that does not need variable
```
Comment (chchhsiao) [inline README.md]: ##### seems not correct, should be #### as well.

**Review by chchhsiao — CHANGES_REQUESTED** [review]

File: README.md [hunk 8]
```diff
@@ -45,91 +61,40 @@ Below is a sample playbook:
...
+install | The install task downloads and installs the Deep Security Agent. The installation is skipped if an agent of the same version is already installed. If a newer version of Deep Security Installer is already installed, then the version is upgraded. | dsm_agent_download_hostname <br/> dsm_agent_download_port
+activate | The activate task activates the Deep Security Agent by registering it in Trend Micro Deep Security Manager. By default, activation is skipped if the agent is already activated, unless the force_reactivation attribute is set to true. | dsm_agent_activation_hostname <br/> dsm_agent_activation_port <br/> policy_id <br/> group_id  <br/> force_reactivation <br/><br/> **For Multi-Tenancy:** <br/> tenant_id <br/> token/tenant_password
+set-policy-by-name | Change agent's policy via policy name | policy_id
+set-policy-by-id | Change agent's policy via policy id | policy_name
+check-in-with-manager | Ask the Agent to contact the Manager now. | ~
```
Comment (chchhsiao) [inline README.md]: For the actions without variable, I'll suggest to use "N/A" in the sample field.

File: README.md [hunk 9]
```diff
@@ -45,91 +61,40 @@ Below is a sample playbook:
 Please refer to the [Variables](#variables) section for required variables.
 
 
-## Variables<a id="variables"></a>
+## Role Variables<a id="variables"></a>
```
Comment (chchhsiao) [inline README.md]: Not sure if Role Variable is a general wording.

File: README.md [hunk 10]
```diff
@@ -45,91 +61,40 @@ Below is a sample playbook:
...
-      action: check-in-with-manager | create-diagnostic-package | ...
-      
-- name: "Set policy for webservers"
-  hosts: web
-  roles:
-    - role: deep-security.deep-security-agent
-      action: set-policy-by-name
-      policy_name: MyCompany - Webserver            
-```
+## Actions Definition
```
Comment (chchhsiao) [inline README.md]: I'll suggest to move the Actions section above the Variable section with a reference link to see the detail.
For users, they'll need to know which action they'd like to use first before setting variables.

File: README.md [hunk 11]
```diff
@@ -45,91 +61,40 @@ Below is a sample playbook:
...
-#### Task : deploy.yml
-###### action: deploy
-
-The <code>deploy</code> task includes the <code>install</code> and <code>activate</code> playbooks internally.
-
-Key | Type | Description | Default
+Key | Type | Description | Sample
 ----|------|-------------|--------
-dsm_agent_download_hostname | String | The hostname of the Deep Security Manager. | app.deepsecurity.trendmicro.com
+action | String | Name of the action to be performed | deploy / install / activate / check-in-with-manager...
```
Comment (chchhsiao) [inline README.md]: The samples could add a reference link to the action section.

**Review by chchhsiao — CHANGES_REQUESTED** [review]

**Comment by harrisonchen201703** [comment]
Hi, @chchhsiao i have updated another version of Readme, may i have your help to take a look on this change? Thanks.

**Review by chchhsiao — APPROVED** [review]

File: README.md [hunk 12]
```diff
@@ -24,8 +24,24 @@ There are no dependencies.
 
 ## Usage
 
-Below is a sample playbook:
+Below are sample usages for playbook:
```
Comment (ChristinaSmith123) [inline README.md]: for playbook -> for the playbook

File: README.md [hunk 13]
```diff
@@ -24,8 +24,24 @@ There are no dependencies.
...
 
+##### For deploying agents to Non-Multi-Tenancy Deep Security Manager
```
Comment (ChristinaSmith123) [inline README.md]: For deploying [...] -> For deploying agents to a non-multi-tenant Deep Security Manager

File: README.md [hunk 14]
```diff
@@ -24,8 +24,24 @@ There are no dependencies.
...
+      dsm_agent_download_hostname: deep.security.manager.host
+      dsm_agent_download_port: 4119
+      dsm_agent_activation_hostname: deep.security.manager.host
+      dsm_agent_activation_port: 4120
+      policy_id: 1
+      group_id: 1
+      force_reactivation: false
+```
+
+##### For deploying agents to Multi-Tenancy Deep Security Manager (like Deep Security as a Service),
```
Comment (ChristinaSmith123) [inline README.md]: For deploying [...]-> For deploying agents to a multi-tenant Deep Security Manager (like Deep Security as a Service)

File: README.md [hunk 15]
```diff
@@ -42,94 +58,43 @@ Below is a sample playbook:
...
-The <code>deploy</code> task includes the <code>install</code> and <code>activate</code> playbooks internally.
-
-Key | Type | Description | Default
+<a id="actions"></a>
+## Actions
+Definition for possible actions to be performed in this role, for required variables please refer to [Variables](#variables).
+
+Action | Description | Variables
+------------|-------------|-------------------
+deploy | The <code>deploy</code> task includes the <code>install</code> and <code>activate</code> playbooks internally. | dsm_agent_download_hostname <br /> dsm_agent_download_port <br /> dsm_agent_activation_hostname	 <br/> dsm_agent_activation_port <br/> policy_id (Optional) <br/> group_id (Optional) <br/> force_reactivation (Optional) <br/><br/> **For Multi-Tenancy:** <br/> tenant_id <br /> token/tenant_password
```
Comment (ChristinaSmith123) [inline README.md]: For Multi-Tenancy: -> For multi-tenancy only:

File: README.md [hunk 16]
```diff
@@ -42,94 +58,43 @@ Below is a sample playbook:
...
+install | The install task downloads and installs the Deep Security Agent. The installation is skipped if an agent of the same version is already installed. If a newer version of Deep Security Installer is already installed, then the version is upgraded. | dsm_agent_download_hostname <br/> dsm_agent_download_port
+activate | The activate task activates the Deep Security Agent by registering it in Trend Micro Deep Security Manager. By default, activation is skipped if the agent is already activated, unless the force_reactivation attribute is set to true. | dsm_agent_activation_hostname <br/> dsm_agent_activation_port <br/> policy_id (Optional) <br/> group_id (Optional) <br/> force_reactivation (Optional) <br/><br/> **For Multi-Tenancy:** <br/> tenant_id <br/> token/tenant_password
```
Comment (ChristinaSmith123) [inline README.md]: For Multi-Tenancy: -> For multi-tenancy only:

File: README.md [hunk 17]
```diff
@@ -42,94 +58,43 @@ Below is a sample playbook:
...
+set-policy-by-name | Change agent's policy via policy name | policy_id
+set-policy-by-id | Change agent's policy via policy id | policy_name
+check-in-with-manager | Ask the Agent to contact the Manager now. | N/A
```
Comment (ChristinaSmith123) [inline README.md]: Agent -> agent
Manager -> manager

File: README.md [hunk 18]
```diff
@@ -42,94 +58,43 @@ Below is a sample playbook:
...
+create-diagnostic-package | Generate an agent diagnostic package. | N/A
+create-integrity-baseline | Rebuilds the integrity monitoring baseline on the computer. | N/A
+run-recommendation-scans | Initiate a recommendation scan on the computer. | N/A
+scan-for-integrity-changes | Scan for changes for Integrity Monitoring | N/A
```
Comment (ChristinaSmith123) [inline README.md]: Integrity Monitoring -> integrity monitoring. (include period too)

File: README.md [hunk 19]
```diff
@@ -42,94 +58,43 @@ Below is a sample playbook:
...
+scan-for-malware | Initiate a manual anti-malware scan | N/A
+update-configuration | Instructs the Deep Security Manager to perform a "Send Policy" operation. | N/A
```
Comment (ChristinaSmith123) [inline README.md]: Instructs -> Instruct

File: README.md [hunk 20]
```diff
@@ -42,94 +58,43 @@ Below is a sample playbook:
...
+install | The install task downloads and installs the Deep Security Agent. The installation is skipped if an agent of the same version is already installed. If a newer version of Deep Security Installer is already installed, then the version is upgraded. | dsm_agent_download_hostname <br/> dsm_agent_download_port
+activate | The activate task activates the Deep Security Agent by registering it in Trend Micro Deep Security Manager. By default, activation is skipped if the agent is already activated, unless the force_reactivation attribute is set to true. | dsm_agent_activation_hostname <br/> dsm_agent_activation_port <br/> policy_id (Optional) <br/> group_id (Optional) <br/> force_reactivation (Optional) <br/><br/> **For Multi-Tenancy:** <br/> tenant_id <br/> token/tenant_password
+set-policy-by-name | Change agent's policy via policy name | policy_id
+set-policy-by-id | Change agent's policy via policy id | policy_name
+check-in-with-manager | Ask the Agent to contact the Manager now. | N/A
+create-diagnostic-package | Generate an agent diagnostic package. | N/A
+create-integrity-baseline | Rebuilds the integrity monitoring baseline on the computer. | N/A
```
Comment (ChristinaSmith123) [inline README.md]: Rebuilds -> Rebuild

File: README.md [hunk 21]
```diff
@@ -42,94 +58,43 @@ Below is a sample playbook:
...
+install | The install task downloads and installs the Deep Security Agent. The installation is skipped if an agent of the same version is already installed. If a newer version of Deep Security Installer is already installed, then the version is upgraded. | dsm_agent_download_hostname <br/> dsm_agent_download_port
+activate | The activate task activates the Deep Security Agent by registering it in Trend Micro Deep Security Manager. By default, activation is skipped if the agent is already activated, unless the force_reactivation attribute is set to true. | dsm_agent_activation_hostname <br/> dsm_agent_activation_port <br/> policy_id (Optional) <br/> group_id (Optional) <br/> force_reactivation (Optional) <br/><br/> **For Multi-Tenancy:** <br/> tenant_id <br/> token/tenant_password
+set-policy-by-name | Change agent's policy via policy name | policy_id
+set-policy-by-id | Change agent's policy via policy id | policy_name
```
Comment (ChristinaSmith123) [inline README.md]: Change agent's -> Change the agent's (change both instances)
via -> via the (change both instances)

File: README.md [hunk 22]
```diff
@@ -42,94 +58,43 @@ Below is a sample playbook:
...
-----|------|-------------|--------
-policy_name | String | The name of the Deep Security policy | 
-
-###### action: set-policy-by-id
+dsm_agent_download_port | Integer | The port to connect to the Deep Security Manager to download the agents. This is typically the same port as the one used to access the Deep Security Manager administration interface. | 443
+force_reactivation | Boolean | Force re-activation even if the Deep Security Agent has already been activated. | false
+group_id | String | The Deep Security ID assigned to the computer group and applied to the agents on activation. | 1
+policy_id | String | ID of the Deep Security Policy to be assigned to the agents. | 1
+policy_name | String | Name of the Deep Security Policy to be assigned to the agents. | Base Policy
+tenant_id | String | In a multi-tenant installation (like Deep Security as a Service), this identifies the tenant account to register the agent with. <br /> **Only required in multi-tenancy environment** | 111A111A-1A1A-11AA-AAA-11AA11111111
```
Comment (ChristinaSmith123) [inline README.md]: Only required in multi-tenancy environment: -> Multi-tenancy only: (to be consistent)

File: README.md [hunk 23]
```diff
@@ -42,94 +58,43 @@ Below is a sample playbook:
...
+token/tenant_password | String | In a multi-tenant installation (like Deep Security as a Service), this identifies the tenant account to register the agent with. In latest Deep Security Manager, "tenant_password" has been replaced with "token". "tenant_password" has been kept for backwards compatibility.<br /> **Only required in multi-tenancy environment** | 111A111A-1A1A-11AA-AAA-11AA11111111
```
Comment (ChristinaSmith123) [inline README.md]: Only required in multi-tenancy environment: -> Multi-tenancy only: (to be consistent)

File: README.md (hunk 12 above)
Comment (harrisonchen201703) [reply README.md]: Thanks, will change in next commit.

**Comment by harrisonchen201703** [comment]
Thanks to @chchhsiao and @ChristinaSmith123 , i will modify the PR again base on these feedback.

File: README.md (hunk 22 above)
Comment (harrisonchen201703) [reply README.md]: To make it looked consistent, do you think it will be better to use "For multi-tenancy only" instead of "Multi-tenancy only"?

**Pull request merged**